From 91891a648a161edd615a0e14b6fb9e4736a1f527 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Tue, 4 Feb 2025 02:06:53 -0500
Subject: [PATCH 19/21] gpg: Default to --compliance=openpgp

* g10/gpg.c (main): default
set_compliance_option(oOpenPGP).
* tests/openpgp/encrypt.scm (Checking OCB): offer --compliance=gnupg
to make it possible to emit LibrePGP AEAD/OCB for the test.
* doc/gpg.texi: change description of what the default compliance
option is.

--

This change is made with the purpose of not inflicting LibrePGP
material on the ecosystem generally unless the user wants it done.

Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
---
 doc/gpg.texi              | 8 ++++----
 g10/gpg.c                 | 2 +-
 tests/openpgp/encrypt.scm | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/doc/gpg.texi b/doc/gpg.texi
index 5a03ce592..5f648e583 100644
--- a/doc/gpg.texi
+++ b/doc/gpg.texi
@@ -3096,15 +3096,15 @@ options.
 @opindex gnupg
 Use standard GnuPG behavior. This is now LibrePGP behavior, which is a
 different draft protocol that overlaps in some cases with
-OpenPGP. This is the default option, so it is not generally needed,
-but it may be useful to override a different compliance option in the
-gpg.conf file.
+OpenPGP.
 
 @item --openpgp
 @opindex openpgp
 Set all packet, cipher and digest options to OpenPGP compatible
 (RFC-9580) behavior.  Note that not all of RFC-9580 is implemented by
-GnuPG.
+GnuPG. This is the default option, so it is not generally needed,
+but it may be useful to override a different compliance option in the
+gpg.conf file.
 
 @item --rfc4880
 @opindex rfc4880
diff --git a/g10/gpg.c b/g10/gpg.c
index 666d9ac3f..4edcb9e97 100644
--- a/g10/gpg.c
+++ b/g10/gpg.c
@@ -2519,7 +2519,7 @@ main (int argc, char **argv)
     opt.passphrase_repeat = 1;
     opt.emit_version = 0;
     opt.weak_digests = NULL;
-    set_compliance_option (oGnuPG);
+    set_compliance_option (oOpenPGP);
 
     /* Check special options given on the command line.  */
     orig_argc = argc;
diff --git a/tests/openpgp/encrypt.scm b/tests/openpgp/encrypt.scm
index ef2f7b0bc..ed903ce5b 100755
--- a/tests/openpgp/encrypt.scm
+++ b/tests/openpgp/encrypt.scm
@@ -88,7 +88,7 @@
  (lambda (source)
    (tr:do
     (tr:open source)
-    (tr:gpgstatus "" `(--yes -e
+    (tr:gpgstatus "" `(--yes -e --compliance=gnupg
                        -r ,"patrice.lumumba"
                        -r ,"mahsa.amini"))
     (tr:call-with-content
-- 
2.50.0

